packages/@okta/vuepress-site/docs/reference/architecture-center/directory-coexistence/lab-tutorial-3-generic-database/index.md

```diff
@@ -0,0 +1,316 @@
+---
+title: Migrate users from an on-premises generic database to Universal Directory
+---
+
+# Directory Coexistence Tutorial 3: Migrate users from an on-premises generic database to Universal Directory
+
+Suppose you store your user profiles and groups in a generic on-premises database but want to replace it with Universal Directory. In this case, you have two options:
+
+1. You export your user information into a CSV (or similar) file and then import the data from the file into Universal Directory.
+1. You write a script that queries the database for user information and then uses the [Okta Users API](/docs/reference/api/users/) to import the users into Universal Directory.
+
+In both cases, Okta can import hashed passwords for users if the hash is one that  [Okta supports](/docs/reference/api/users/#hashed-password-object). If Okta does not support the hash, it will not import the user's password. However, you can use the [Okta Password Inline Import Hook](/docs/reference/password-hook/) to import user records that use an unsupported hash type, without requiring that the user resets their password.
```

Comment:
```suggestion
In both cases, Okta can import hashed passwords for users if the hash is one that  [Okta supports](/docs/reference/api/users/#hashed-password-object). If Okta does not support the hash, it will not import the user's password. However, you can use the [Okta Password Inline Import Hook](/docs/reference/password-hook/) to import user records that use an unsupported hash type, without requiring that the user resets their password.
```
```suggestion
In both cases, Okta can import hashed passwords for users if the hash is one that  [Okta supports](/docs/reference/api/users/#hashed-password-object). If Okta does not support the hash, it does not import the user's password. However, you can use the [Okta Password Inline Import Hook](/docs/reference/password-hook/) to import user records that use an unsupported hash type, without requiring that the user resets their password.
```